pkg/services/ngalert/schedule/schedule.go

```diff
@@ -307,12 +318,12 @@ func (sch *schedule) adminConfigSync(ctx context.Context) error {
 			}
 		case <-ctx.Done():
 			// Stop sending alerts to all external Alertmanager(s).
-			sch.sendersMtx.Lock()
+			sch.adminConfigMtx.Lock()
```

Comment:
I know this was just renamed, but isn't this a deadlock? We cannot acquire an exclusive lock on `adminConfigMtx` when we also have a shared lock on it from Line 302?